apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.Data/SpannerCommandTextBuilder.cs

```diff
@@ -0,0 +1,195 @@
+            return new SpannerCommandTextBuilder {
+                SpannerCommandType = SpannerCommandType.Insert,
+                TargetTable = databaseTableName,
+                CommandText = $"{s_insertCommand} {databaseTableName}"
+            };
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="sqlQuery"></param>
+        /// <returns></returns>
+        public static SpannerCommandTextBuilder CreateSelectTextBuilder(string sqlQuery)
+        {
+            return new SpannerCommandTextBuilder {
+                SpannerCommandType = SpannerCommandType.Select,
+                CommandText = sqlQuery
+            };
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="databaseTableName"></param>
+        /// <returns></returns>
+        public static SpannerCommandTextBuilder CreateUpdateTextBuilder(string databaseTableName)
+        {
+            ValidateTable(databaseTableName);
+            return new SpannerCommandTextBuilder {
+                SpannerCommandType = SpannerCommandType.Update,
+                TargetTable = databaseTableName,
+                CommandText = $"{s_updateCommand} {databaseTableName}"
+            };
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="commandText"></param>
+        /// <returns></returns>
+        public static SpannerCommandTextBuilder FromCommandText(string commandText)
+        {
+            GaxPreconditions.CheckNotNullOrEmpty(commandText, nameof(commandText));
+            return new SpannerCommandTextBuilder(commandText);
+        }
+
+        /// <inheritdoc />
+        public override string ToString()
+        {
+            return CommandText;
+        }
+
+        private static bool TryParseCommand(SpannerCommandTextBuilder newbuilder,
+            string commandToParseFor, SpannerCommandType commandType,
+            string[] commandSections)
+        {
+            var operationName = commandSections[0].ToUpper();
+            if (Equals(operationName, commandToParseFor))
+            {
+                if (commandSections.Length != 2)
+                    throw new InvalidOperationException(
+                        $"Spanner {commandToParseFor} commands are specified as '{commandToParseFor} <table>' with " +
+                        "parameters added to customize the command with filtering or updated values.");
+                newbuilder.CommandText = $"{commandToParseFor} {commandSections[1]}";
+                newbuilder.SpannerCommandType = commandType;
+                newbuilder.TargetTable = commandSections[1];
+                return true;
+            }
+            return false;
+        }
+
+        private static void ValidateTable(string databaseTableName)
+        {
+            GaxPreconditions.CheckNotNullOrEmpty(databaseTableName, nameof(databaseTableName));
+            if (!databaseTableName.All(c => char.IsLetterOrDigit(c) || c == '_'))
```

Comment:
would prefer not to use regex for a check as simple as letter+digit+underscore.
in my experience regex's always have a perf hit.